src/core/diagnostictool.cpp

```diff
@@ -236,3 +236,26 @@ unsigned int DiagnosticTool::getAmountOfResolvers()
 {
     return d->resolvers->size();
 }
+
+bool DiagnosticTool::anyErrorsInChecks()
+{
+    auto &checks = (*d->checks.get());
+
+    for (auto &check : checks)
+    {
+        if (check.get()->m_exit_code != 0)
+        {
+            return true;
+        }
+    }
+
+    return false;
+}
+
+bool DiagnosticTool::hasAnyResolvers()
+{
+    if(d->resolvers->size() == 0) {
```

Comment:
You can write this as:
return !d->resolvers->empty();

Reply:
fixed